crates/services/txpool/src/error.rs

```diff
@@ -0,0 +1,166 @@
+use fuel_core_types::{
+    fuel_tx::{
+        Address,
+        BlobId,
+        ContractId,
+        TxId,
+        UtxoId,
+    },
+    fuel_types::Nonce,
+    fuel_vm::checked_transaction::CheckError,
+};
+
+use crate::ports::WasmValidityError;
+
+pub type Result<T> = core::result::Result<T, Error>;
+
+#[allow(missing_docs)]
+#[derive(Debug, Clone, derive_more::Display)]
+#[non_exhaustive]
+pub enum Error {
+    #[display(fmt = "Gas price not found for block height {_0}")]
+    GasPriceNotFound(String),
+    #[display(fmt = "Failed to calculate max block bytes: {_0}")]
+    MaxBlockBytes(String),
+    #[display(fmt = "TxPool required that transaction contains metadata")]
+    NoMetadata,
+    #[display(fmt = "TxPool doesn't support this type of transaction.")]
+    NotSupportedTransactionType,
+    #[display(fmt = "Transaction is not inserted. Hash is already known")]
+    NotInsertedTxKnown,
+    #[display(
+        fmt = "Transaction is not inserted. Pool limit is hit, try to increase gas_price"
+    )]
+    NotInsertedLimitHit,
+    #[display(
+        fmt = "Transaction is not inserted. Attempting to upgrade WASM bytecode when WASM support is not enabled."
+    )]
+    NotInsertedWasmNotEnabled,
+    #[display(
+        fmt = "Transaction is not inserted. WASM bytecode matching the given root was not found."
+    )]
+    NotInsertedWasmNotFound,
+    #[display(
+        fmt = "Transaction is not inserted. WASM bytecode contents are not valid."
+    )]
+    NotInsertedInvalidWasm,
+    #[display(fmt = "Transaction is not inserted. The gas price is too low.")]
+    NotInsertedGasPriceTooLow,
+    #[display(
+        fmt = "Transaction is not inserted. More priced tx {_0:#x} already spend this UTXO output: {_1:#x}"
+    )]
+    NotInsertedCollision(TxId, UtxoId),
+    #[display(
+        fmt = "Transaction is not inserted. More priced tx has created contract with ContractId {_0:#x}"
```

Comment:
```suggestion
        fmt = "Transaction is not inserted. Higher priced tx has created contract with ContractId {_0:#x}"
```

Reply:
perhaps we also want to add the tx hash for the already spent utxo like the above error :)

Reply:
From what I can see, we only have `ContractId` here so I believe you mistook this for the previous `NotInsertedCollision` variant.

Reply:
But I've updated the message ✅

Reply:
yup, I meant to ask if we should include the higher priced tx hash like we have done in other error cases, what do you think?

Reply:
Oh sorry I misunderstood you. Yeah that sounds helpful. I'll see if I can include it.

Reply:
Getting the txid here doesn't look trivial based on my limited understanding of the code unfortunately.

Reasoning:
This variant is only created in `crates/services/txpool/src/containers/dependency.rs` in a loop that looks like this:

```
        for output in tx.outputs() {
            if let Output::ContractCreated { contract_id, .. } = output {
                if let Some(contract) = self.contracts.get(contract_id) {
                    // we have a collision :(
                    if contract.is_in_database() {
                        return Err(Error::NotInsertedContractIdAlreadyTaken(*contract_id))
                    }
                    // check who is priced more
                    if contract.tip > tx.tip() {
                        // new tx is priced less then current tx
                        return Err(Error::NotInsertedCollisionContractId(*contract_id))
                    }
                    // if we are prices more, mark current contract origin for removal.
                    let origin = contract.origin.expect(
                        "Only contract without origin are the ones that are inside DB. And we check depth for that, so we are okay to just unwrap"
                        );
                    collided.push(*origin.tx_id());
                }
            }
            // collision of other outputs is not possible.
        }
```

The `contract` variable here is of the following type
```
#[derive(Debug, Clone)]
pub struct ContractState {
    /// is Utxo spend as other Tx input
    used_by: HashSet<TxId>,
    /// how deep are we inside UTXO dependency
    depth: usize,
    /// origin is needed for child to parent rel, in case when contract is in dependency this is how we make a chain.
    origin: Option<UtxoId>,
    /// gas_price. We can probably derive this from Tx
    tip: Word,
}
```

which does not seem to hold the information about which txid created the contract.

There may be other ways to fetch the txid, but that would complect the code for what seems to be a very unusual error scenario so I'll leave this as-is for now if that's alright with you.